stable_baselines3/common/vec_env/base_vec_env.py

```diff
@@ -82,12 +66,12 @@ def reset(self):
         be cancelled and step_wait() should not be called
         until step_async() is invoked again.
 
-        :return: ([int] or [float]) observation
+        :return: (VecEnvObs) observation
         """
         raise NotImplementedError()
 
     @abstractmethod
-    def step_async(self, actions):
+    def step_async(self, actions: np.ndarray):
```

Comment:
Hmm this might be reverted when we start implementing support for dict/tuple action-spaces, where we can not stack actions into a numpy array. This feature is considered to be added right after v1.0, being so well requested. Same applies to observations.

Reply:
I'm not so confident about dict/tuple support for action spaces, sounds like an open research question to me (mixing proba distributions)

for observations, this is already included in the type ;)

Reply:
Actions in those cases are usually treated independent, as they are independent in the environment specifications, but indeed it is not super clear how it should be done. Lets go with the current modifications and deal with that later.